packages/adena-extension/src/hooks/web/common/use-create-password-screen.ts

```diff
@@ -0,0 +1,183 @@
+  buttonState: {
+    onClick: () => void;
+    disabled: boolean;
+  };
+  onKeyDown: (e: React.KeyboardEvent<HTMLInputElement>) => void;
+};
+
+export const useCreatePasswordScreen = (): UseCreatePasswordScreenReturn => {
+  const { updateWallet } = useWalletContext();
+  const { walletService } = useAdenaContext();
+  const { navigate, params } = useAppNavigate<RoutePath.WebCreatePassword>();
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const [inputs, setInputs] = useState({
+    password: '',
+    confirmPassword: '',
+  });
+  const [terms, setTerms] = useState(false);
+  const [isPasswordError, setIsPasswordError] = useState(false);
+  const [isConfirmPasswordError, setIsConfirmPasswordError] = useState(false);
+  const { password, confirmPassword } = inputs;
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const disabledCreateButton = useMemo(() => {
+    return !terms || !password || !confirmPassword;
+  }, [terms, password, confirmPassword]);
+
+  const _validateConfirmPassword = (validationPassword?: boolean): boolean => {
+    try {
+      if (validateNotMatchConfirmPassword(password, confirmPassword)) return true;
+    } catch (error) {
+      if (error instanceof PasswordValidationError) {
+        switch (error.getType()) {
+          case 'NOT_MATCH_CONFIRM_PASSWORD':
+            setIsConfirmPasswordError(true);
+            break;
+          default:
+            break;
+        }
+        if (validationPassword) {
+          setErrorMessage(error.message);
+        }
+      }
+    }
+    return false;
+  };
+
+  const _validateInputPassword = (): boolean => {
+    try {
+      validateEmptyPassword(password);
+      validateWrongPasswordLength(password);
+      return true;
+    } catch (error) {
+      console.log(error);
+      setIsPasswordError(true);
+      if (error instanceof PasswordValidationError) {
+        setErrorMessage(error.message);
+      }
+    }
+    return false;
+  };
+
+  const validateMatchPassword = (): boolean => {
+    const validatedInputPassword = _validateInputPassword();
+    const validatedConfirmPassword = _validateConfirmPassword(validatedInputPassword);
+
+    try {
+      if (validatedInputPassword && validatedConfirmPassword) {
+        return true;
+      }
+    } catch (error) {
+      console.error(error);
+    }
+    return false;
+  };
+
+  const _saveWalletByPassword = async (password: string): Promise<void> => {
+    const { serializedWallet } = params;
+    const wallet = await AdenaWallet.deserialize(serializedWallet, '');
+    await walletService.saveWallet(wallet, password);
+    await updateWallet(wallet);
+  };
+
+  const onChangePassword = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const { name, value } = e.target;
+      setInputs((input) => ({ ...input, [name]: value }));
+    },
+    [setInputs],
+  );
+
+  const toggleTermState = useCallback((): void => {
+    setTerms((prev: boolean) => !prev);
+  }, [setTerms]);
```

Comment:
```suggestion
setTerms(!terms);
  }, [terms]);
```

set function is immutable